### PR TITLE
feat: Composable CMS components

### LINKED ISSUE #12346: Composable CMS components
Provide a solution for the fine-grained extensibility of lower-level components.
Such a solution should allow for runtime and configuration-driven composable components, i.e. components that can render other subcomponents.

### PR DESCRIPTION
**Pull request by dunqan**
Provides a solution for the fine-grained extensibility of lower-level components.
Allows for runtime and configuration-driven composable components, i.e. components that can render other subcomponents.

Closes #12346

### CODE REVIEW

File: projects/storefrontlib/src/cms-structure/page/component/page-component.module.ts [hunk 1]
```diff
@@ -19,7 +20,7 @@ import { LazyComponentHandler } from './handlers/lazy-component.handler';
       multi: true,
     },
   ],
```
Comment (Platonn) [inline projects/storefrontlib/src/cms-structure/page/component/page-component.module.ts:22]: Shouldn't we move above providers to forRoot() method?

File: projects/storefrontlib/src/cms-structure/page/component/inner-components-host.directive.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    protected cmsComponentsService: CmsComponentsService,
+    protected injector: Injector,
+    protected dynamicAttributeService: DynamicAttributeService,
+    protected renderer: Renderer2,
+    protected componentHandler: ComponentHandlerService,
+    protected cmsInjector: CmsInjectorService
+  ) {}
+
+  ngOnInit(): void {
+    this.innerComponents$.subscribe((x) => {
```
Comment (Platonn) [inline projects/storefrontlib/src/cms-structure/page/component/inner-components-host.directive.ts]:
```suggestion
    this.subscription = this.innerComponents$.subscribe((x) => {
```

Comment (dunqan) [reply projects/storefrontlib/src/cms-structure/page/component/inner-components-host.directive.ts]: Thanks!

File: projects/storefrontlib/src/cms-structure/page/component/page-component.module.ts (hunk 1 above)
Comment (dunqan) [reply projects/storefrontlib/src/cms-structure/page/component/page-component.module.ts:22]: Interesting observation. As it's a breaking change, it requires separate ticket (for 4.0 probably)

**Review by Platonn — DISMISSED** [review]

**Review by Platonn — APPROVED** [review]

**Pull request merged**